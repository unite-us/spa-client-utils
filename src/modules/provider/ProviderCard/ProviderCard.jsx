import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { get, find, isEmpty, map, noop, orderBy } from 'lodash';
import { Icon } from '@unite-us/ui';
import serviceTypeUtils from 'modules/utils/serviceTypes';
import addresses from 'modules/utils/addresses';
import constants from 'modules/utils/constants';
import arrays from 'modules/utils/arrays';
import phones from 'modules/utils/phones';
import urls from 'modules/utils/urls';
import { DAYS_OF_THE_WEEK, CLOSED_DAY } from 'modules/hours/HoursOfOperation/constants';
import { formatDuration } from 'modules/hours/HoursOfOperation/utils';
import EmailAddress from 'modules/contact/EmailAddress';
import ProviderCardAddRemoveButton from './components/ProviderCardAddRemoveButton';

class ProviderCard extends Component {
  constructor(props) {
    super(props);

    this.onAddProvider = this.onAddProvider.bind(this);
    this.onRemoveProvider = this.onRemoveProvider.bind(this);
    this.onDetailClick = this.onDetailClick.bind(this);
  }

  onAddProvider(destinationLatLng) {
    const {
      provider,
      onAddProvider,
      originLatLng,
    } = this.props;

    const distance = addresses.getDistanceInMiles(originLatLng, destinationLatLng, {
      format: 'float',
    });

    const providerWithDistance = { ...provider, distance };

    onAddProvider(providerWithDistance);
  }

  onRemoveProvider() {
    const { provider, onRemoveProvider } = this.props;
    onRemoveProvider(provider);
  }

  onDetailClick() {
    const { provider, onDetailClick } = this.props;
    onDetailClick(provider);
  }

  render() {
    const {
      className,
      disableSelection,
      isSelected,
      onDetailClick,
      originLatLng,
      provider,
      selectedServiceType,
      unselectable,
    } = this.props;

    const date = new Date();
    const today = DAYS_OF_THE_WEEK[date.getDay() - 1];
    const hasHoursOfOperation = !isEmpty(provider.hours_of_operation) || !isEmpty(provider.hours);
    let hoursDisplay = '';
    let todayHours = [];
    if (provider.hours_of_operation !== undefined) {
      todayHours = find(provider.hours_of_operation, { day_of_week: today }) || {};
      hoursDisplay = map(
        orderBy(todayHours.hours_of_operation, 'opens_at'),
        formatDuration,
      ).join(', ');
    } else {
      todayHours = get(provider, ['hours', today], []);
      hoursDisplay = map(
        todayHours,
        formatDuration,
      ).join(', ');
    }

    const email = arrays.findPrimaryOrFirst(provider.email_addresses);
    const phone = arrays.findPrimaryOrFirst(provider.phone_numbers);

    const destinationLatLng = provider.addresses[0] && provider.addresses[0].lat_lng !== undefined ?
      get(provider, 'addresses[0].lat_lng', []) :
      [get(provider, 'addresses[0].latitude', []), get(provider, 'addresses[0].longitude', [])];
    const distance = addresses.getDistanceInMiles(originLatLng, destinationLatLng);

    const detailsProps = !onDetailClick ? {} : {
      onClick: this.onDetailClick,
      onKeyPress: this.onDetailClick,
      role: 'button',
      style: { cursor: 'pointer' },
      tabIndex: 0,
    };

    const isInNetworkGroup = get(provider, 'group_type') === constants.IN_NETWORK_GROUP_TYPE || get(provider, 'licensed') === true;
    const providerServiceTypes = provider.service_types;

    const serviceTypes = serviceTypeUtils.matchingServiceTypes(selectedServiceType, providerServiceTypes);
    const svcTypeBanner = serviceTypeUtils.serviceTypeBanner(serviceTypes);

    const addressString = addresses.getAddressString(get(provider, 'addresses[0]'));

    return (
      <div className={classnames('ui-provider-card', className)}>
        <div className={
          classnames('ui-provider-card__info', {
            'ui-provider-card__info-disabled': disableSelection,
          })
        }
        >
          <div className="ui-provider-card__detail" {...detailsProps}>
            {
              (!isInNetworkGroup) &&
                <div className="ui-provider-card__oon">
                  <Icon
                    icon="IconElectronsArrow"
                    size={20}
                    className="mr-half"
                  />
                  <span>
                    Out of Network Organization
                  </span>
                </div>
            }
            <h4 className="ui-provider-card__name">{provider.name}</h4>
            {
              !isEmpty(svcTypeBanner) &&
                <p className="ui-provider-card__service-type">
                  {svcTypeBanner}
                </p>
            }
            {
              !isEmpty(provider.addresses) &&
              <p
                className="ui-provider-card__address mb-quarter"
                title={addressString}
              >
                <Icon icon="IconMapMarker" size={16} />
                {addressString}
              </p>
            }
            {
              !isEmpty(distance) &&
              <p className="ui-provider-card__distance mb-quarter">
                <span><Icon icon="IconCompass" size={16} /></span>
                <span>{distance}</span>
              </p>
            }
            {
              hasHoursOfOperation &&
              <p className="ui-provider-card__hours mb-quarter">
                <span><Icon icon="IconClock" size={16} /></span>
                <span>TODAY:&nbsp;</span>
                <span>
                  {isEmpty(todayHours) ? CLOSED_DAY : hoursDisplay}
                </span>
              </p>
            }
            {
              !isEmpty(phone) &&
              <p className="ui-provider-card__phone mb-quarter">
                <a
                  href={phones.getTelLink(phone)}
                  target="_blank"
                  tabIndex={0}
                  rel="noopener noreferrer"
                >
                  <span><Icon icon="IconPhone" size={12} /></span>
                  {phones.formatPhoneNumber(phone.extension ?
                    phone.phone_number + phone.extension : phone.phone_number)}
                </a>
              </p>
            }
            {
              /** @todo Do we need a way to hide emails in emr? */
              !isEmpty(email) &&
              <p className="ui-provider-card__email mb-quarter">
                <span>
                  <Icon icon="IconEnvelope" size={16} />
                  <EmailAddress emailAddress={email.email_address} />
                </span>
              </p>
            }
            {
              provider.website_url &&
              <p className="ui-provider-card__url mb-quarter">
                <a
                  href={urls.addHttpIfNeeded(provider.website_url)}
                  target="_blank"
                  rel="noopener noreferrer"
                  tabIndex={0}
                >
                  <span><Icon icon="IconWeb" size={16} /></span>
                  {provider.website_url}
                </a>
              </p>
            }
          </div>
        </div>
        {
          !disableSelection && !unselectable &&
            <ProviderCardAddRemoveButton
              selected={isSelected}
              distance={distance}
              destinationLatLng={destinationLatLng}
              onAddClick={this.onAddProvider}
              onRemoveClick={this.onRemoveProvider}
            />
        }
      </div>
    );
  }
}

ProviderCard.propTypes = {
  /** className */
  className: PropTypes.string,
  /** true for disabled card state, hidden add/remove button */
  disableSelection: PropTypes.bool,
  /** true indicates this provider is selected */
  isSelected: PropTypes.bool,
  /** called on add button click */
  onAddProvider: PropTypes.func,
  /** called on provider card click */
  onDetailClick: PropTypes.func,
  /** called on remove button click */
  onRemoveProvider: PropTypes.func,
  /** center of map to calculate distance */
  originLatLng: PropTypes.array,
  /** object containing provider data */
  provider: PropTypes.shape({
    addresses: PropTypes.array,
    email_addresses: PropTypes.array,
    group_type: PropTypes.string,
    hours: PropTypes.object,
    hours_of_operation: PropTypes.array,
    name: PropTypes.string,
    phone_numbers: PropTypes.array,
    programs: PropTypes.array,
    service_types: PropTypes.array,
    website_url: PropTypes.string,
  }).isRequired,
  /** service type object or array of service type objects */
  selectedServiceType: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.shape({ id: PropTypes.string, name: PropTypes.string }),
    ),
    PropTypes.shape({ id: PropTypes.string, name: PropTypes.string }),
  ]).isRequired,
  /** true to hide add/remove buttons */
  unselectable: PropTypes.bool,
};

ProviderCard.defaultProps = {
  className: '',
  disableSelection: false,
  isSelected: false,
  onAddProvider: noop,
  onDetailClick: noop,
  onRemoveProvider: noop,
  originLatLng: [],
  unselectable: false,
};

export default ProviderCard;
