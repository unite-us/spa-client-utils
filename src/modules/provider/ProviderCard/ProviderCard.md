```jsx
initialState = {
  isSelected: false,
};

const hours_of_operation = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday']
  .map(day_of_week => ({
    day_of_week,
    hours_of_operation: [{ opens_at: 540, closes_at: 1020 }],
  })
);

const hours = {
  monday: [{ opens: '08:00', closes: '17:00' }],
  tuesday: [{ opens: '08:00', closes: '17:00' }],
  wednesday: [{ opens: '08:00', closes: '12:00' }, { opens: '14:00', closes: '17:00' }],
  thursday: [{ opens: '08:00', closes: '17:00' }],
  friday: [{ opens: '08:00', closes: '17:00' }],
};

<ProviderCard
  isSelected={state.isSelected}
  onAddProvider={() => { setState({ isSelected: true }) }}
  onRemoveProvider={() => { setState({ isSelected: false }) }}
  originLatLng={[42.4191639, -71.0269413]}
  provider={{
    addresses: [{
      line_1: '123 Loooooooooooooooooooong Street',
      city: 'Funmouth',
      state: 'MA',
      postal_code: '12345',
      country: 'USA',
      address_type: 'work',
      lat_lng: [42.3682235, -71.2398932],
    }],
    email_addresses: [
      { email_address: 'first@email.com' },
      { email_address: 'primary@email.com', is_primary: true },
    ],
    group_type: 'out_of_network',
    hours_of_operation,
    hours,
    name: 'Some Organization',
    phone_numbers: [{
      phone_number: '1234567890',
      is_primary: true,
      phone_type: 'work',
      can_sms: true,
    }],
    programs: [{ id: 'program-id', name: 'Some Program' }],
    service_types: [{ id: 'service-type-id', name: 'Some Service Type' }],
    website_url: 'www.thisorgsite.com',
  }}
  selectedServiceType={{ id: 'service-type-id', name: 'Some Service Type' }}
/>
```
