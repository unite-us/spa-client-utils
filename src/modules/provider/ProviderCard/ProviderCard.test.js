import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import group from 'testUtils/group.json';
import ProviderCard from './ProviderCard';

jest.mock('@unite-us/ui', () => ({
  ...require.requireActual('@unite-us/ui'),
  Icon: jest.fn(() => <mock-icon />),
}));

// TODO: fix tests for day of week
describe('ProviderCard', () => {
  const mockDate = new Date('2020-01-01T00:00:00');

  beforeEach(() => {
    global.Date = class extends Date {
      constructor() {
        super();
        return mockDate;
      }
    };
  });

  it('renders', () => {
    const props = {
      provider: group,
      selectedServiceType: {},
    };

    expect(renderer.create(<ProviderCard {...props} />)).toMatchSnapshot();
  });

  it('calls the add/remove callbacks', () => {
    const onAddProvider = jest.fn();
    const onRemoveProvider = jest.fn();
    const props = {
      disableSelection: false,
      provider: group,
      isSelected: false,
      onAddProvider,
      onRemoveProvider,
      selectedServiceType: {},
    };

    const groupWithDistance = {
      ...group,
      distance: '',
    };

    const comp = shallow(<ProviderCard {...props} />);
    comp.instance().onAddProvider();
    expect(onAddProvider).toHaveBeenCalledWith(groupWithDistance);
    comp.instance().onRemoveProvider();
    expect(onRemoveProvider).toHaveBeenCalledWith(group);
  });

  it('renders multiple service types', () => {
    const props = {
      provider: group,
      selectedServiceType: [
        {
          id: '50126fcb-a3e9-45d5-9a8e-c32b7fd810b4',
          name: 'Benefits',
        },
        {
          id: 'c088b0bd-9d90-4aaf-9841-ff29bf6311b3',
          name: 'Clothing & Household Goods',
        },
        {
          id: '123-ba-match',
          name: 'No Match Here',
        },
      ],
    };

    expect(renderer.create(<ProviderCard {...props} />)).toMatchSnapshot();
  });

  it('renders with no service types', () => {
    const props = {
      provider: group,
      selectedServiceType: [],
    };

    expect(renderer.create(<ProviderCard {...props} />)).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const props = {
      provider: group,
      selectedServiceType: [],
    };

    expect(renderer.create(
      <ProviderCard {...props} disableSelection />,
    )).toMatchSnapshot();
  });

  it('renders unselectable', () => {
    const props = {
      provider: group,
      selectedServiceType: [],
    };
    expect(renderer.create(
      <ProviderCard {...props} unselectable />,
    )).toMatchSnapshot();
  });

  it('calls the onDetailClick callback', () => {
    const onDetailClick = jest.fn();
    const props = {
      disableSelection: false,
      provider: group,
      isSelected: false,
      onAddProvider: jest.fn(),
      onRemoveProvider: jest.fn(),
      selectedServiceType: {},
      onDetailClick,
    };

    const event = { preventDefault: jest.fn() };

    const comp = shallow(<ProviderCard {...props} />);
    comp.instance().onDetailClick(event);
    expect(onDetailClick).toHaveBeenCalledWith(group);
  });

  it('shows indication for OON provider', () => {
    const props = {
      provider: {
        ...group,
        group_type: 'out_of_network',
      },
      selectedServiceType: {},
    };

    expect(renderer.create(<ProviderCard {...props} />)).toMatchSnapshot();
  });

  it('shows indication for external providers (same as OON)', () => {
    const props = {
      provider: {
        ...group,
        group_type: 'external_provider',
      },
      selectedServiceType: {},
    };

    expect(renderer.create(<ProviderCard {...props} />)).toMatchSnapshot();
  });

  describe('provider details', () => {
    const defaultProps = {
      isSelected: false,
      onAddProvider: jest.fn(),
      onDetailClick: jest.fn(),
      onRemoveProvider: jest.fn(),
      originLatLng: [42.4191639, -71.0269413],
      provider: group,
      selectedServiceType: {},
    };

    it('hides address when provider address does not exist', () => {
      const props = {
        ...defaultProps,
        provider: {
          ...group,
          addresses: [],
        },
      };

      const wrapper = shallow(<ProviderCard {...props} />);
      expect(wrapper.find('.ui-provider-card__address')).toHaveLength(0);
    });

    it('hides distance when provider address does not exist', () => {
      const wrapper = shallow(<ProviderCard {...defaultProps} originLatLng={[]} />);
      expect(wrapper.find('.ui-provider-card__distance')).toHaveLength(0);
    });

    it('hides distance when originLatLng does not exist', () => {
      const wrapper = shallow(<ProviderCard {...defaultProps} originLatLng={[]} />);
      expect(wrapper.find('.ui-provider-card__distance')).toHaveLength(0);
    });

    it('hides hours of operation when provider hours_of_operation does not exist', () => {
      const props = {
        ...defaultProps,
        provider: {
          ...group,
          hours_of_operation: [],
          hours: {},
        },
      };

      const wrapper = shallow(<ProviderCard {...props} />);
      expect(wrapper.find('.ui-provider-card__hours')).toHaveLength(0);
    });

    it('hides phone number when provider phone_numbers does not exist', () => {
      const props = {
        ...defaultProps,
        provider: {
          ...group,
          phone_numbers: [],
        },
      };

      const wrapper = shallow(<ProviderCard {...props} />);
      expect(wrapper.find('.ui-provider-card__phone')).toHaveLength(0);
    });

    it('hides website url when provider website_url does not exist', () => {
      const props = {
        ...defaultProps,
        provider: {
          ...group,
          website_url: '',
        },
      };

      const wrapper = shallow(<ProviderCard {...props} />);
      expect(wrapper.find('.ui-provider-card__url')).toHaveLength(0);
    });

    it('Show provider hours of operation', () => {
      const props = {
        ...defaultProps,
        provider: {
          ...group,
        },
      };

      const wrapper = shallow(<ProviderCard {...props} />);
      expect(wrapper.find('.ui-provider-card__hours')).toHaveLength(1);
      expect(wrapper.find('.ui-provider-card__hours').text().includes('9:00 AM - 5:00 PM')).toBe(true);
    });
  });
});
