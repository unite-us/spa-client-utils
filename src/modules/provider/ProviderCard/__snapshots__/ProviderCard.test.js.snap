// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderCard renders 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
  <div
    className="provider-card-add-remove-button"
  >
    <div
      className="ui-add-remove-buttons"
    >
      <div
        className="ui-add-remove-buttons__add"
      >
        <a
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Add"
        >
          <mock-icon />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderCard renders disabled 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info ui-provider-card__info-disabled"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`ProviderCard renders multiple service types 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
  <div
    className="provider-card-add-remove-button"
  >
    <div
      className="ui-add-remove-buttons"
    >
      <div
        className="ui-add-remove-buttons__add"
      >
        <a
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Add"
        >
          <mock-icon />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderCard renders unselectable 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`ProviderCard renders with no service types 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
  <div
    className="provider-card-add-remove-button"
  >
    <div
      className="ui-add-remove-buttons"
    >
      <div
        className="ui-add-remove-buttons__add"
      >
        <a
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Add"
        >
          <mock-icon />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderCard shows indication for OON provider 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <div
        className="ui-provider-card__oon"
      >
        <mock-icon />
        <span>
          Out of Network Organization
        </span>
      </div>
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
  <div
    className="provider-card-add-remove-button"
  >
    <div
      className="ui-add-remove-buttons"
    >
      <div
        className="ui-add-remove-buttons__add"
      >
        <a
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Add"
        >
          <mock-icon />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProviderCard shows indication for external providers (same as OON) 1`] = `
<div
  className="ui-provider-card"
>
  <div
    className="ui-provider-card__info"
  >
    <div
      className="ui-provider-card__detail"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      tabIndex={0}
    >
      <div
        className="ui-provider-card__oon"
      >
        <mock-icon />
        <span>
          Out of Network Organization
        </span>
      </div>
      <h4
        className="ui-provider-card__name"
      >
        Testing Org [DEV only]
      </h4>
      <p
        className="ui-provider-card__address mb-quarter"
        title="10 Columbus Circle, New York, NY 10019"
      >
        <mock-icon />
        10 Columbus Circle, New York, NY 10019
      </p>
      <p
        className="ui-provider-card__hours mb-quarter"
      >
        <span>
          <mock-icon />
        </span>
        <span>
          TODAY: 
        </span>
        <span>
          9:00 AM - 5:00 PM
        </span>
      </p>
      <p
        className="ui-provider-card__phone mb-quarter"
      >
        <a
          href="tel:+12128921010"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          (212) 892-1010
        </a>
      </p>
      <p
        className="ui-provider-card__email mb-quarter"
      >
        <span>
          <mock-icon />
          <span
            className="ui-email-address"
          >
            <a
              href="mailto:test@manager.com"
              title="test@manager.com"
            >
              test@manager.com
            </a>
          </span>
        </span>
      </p>
      <p
        className="ui-provider-card__url mb-quarter"
      >
        <a
          href="http://www.app.uniteusdev.com/"
          rel="noopener noreferrer"
          tabIndex={0}
          target="_blank"
        >
          <span>
            <mock-icon />
          </span>
          http://www.app.uniteusdev.com/
        </a>
      </p>
    </div>
  </div>
  <div
    className="provider-card-add-remove-button"
  >
    <div
      className="ui-add-remove-buttons"
    >
      <div
        className="ui-add-remove-buttons__add"
      >
        <a
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Add"
        >
          <mock-icon />
        </a>
      </div>
    </div>
  </div>
</div>
`;
